### PR TITLE
[EOC-245] Dispaly confirmation on changing sack type

### PR DESCRIPTION
**Pull request by alemikolo**
Screenshot 2019-06-27_13-45-43-157

### CODE REVIEW

File: client/src/modules/list/components/ListHeader.jsx [hunk 1]
```diff
@@ -16,22 +16,26 @@ import DescriptionTextarea from 'common/components/DescriptionTextarea';
...
 
     this.state = {
       descriptionInputValue: trimmedDescription,
       isDescriptionTextareaVisible: false,
+      isDialogVisible: false,
       isNameInputVisible: false,
       isTipVisible: false,
+      listType: type,
       nameInputValue: name,
+      newType: null,
```
Comment (adamklepacz) [inline client/src/modules/list/components/ListHeader.jsx]: updatedType: null

File: client/src/modules/list/components/ListHeader.jsx [hunk 2]
```diff
@@ -182,23 +186,22 @@ class ListHeader extends PureComponent {
...
-    } = event;
+  handleChangingType = () => {
     const {
       changeType,
       details: { name },
       match: {
         params: { id: listId }
       }
     } = this.props;
+    const { newType } = this.state;
```
Comment (adamklepacz) [inline client/src/modules/list/components/ListHeader.jsx]: updatedType

File: client/src/modules/list/components/ListHeader.jsx [hunk 3]
```diff
@@ -296,17 +309,16 @@ class ListHeader extends PureComponent {
...
+    const { listType, pendingForType } = this.state;
 
     return (
       <select
         className="list-header__select primary-select"
-        defaultValue={type}
+        value={listType}
         disabled={pendingForType}
-        onChange={this.handleChangingType}
+        onChange={this.showDialog}
```
Comment (adamklepacz) [inline client/src/modules/list/components/ListHeader.jsx]: sort props

File: client/src/locales/en.json [hunk 4]
```diff
@@ -152,7 +152,9 @@
   "list.index.hide-members": "Hide sacks members",
   "list.index.show-members": "Show sacks members",
   "list.index.archivization": "'{name}' sack archivization...",
+  "list.index.changing-type": "Changing '{name}' sack's type  to {value}...",
```
Comment (marekrozmus) [inline client/src/locales/en.json]: double space after `type`

File: client/src/locales/en.json [hunk 5]
```diff
@@ -152,7 +152,9 @@
...
   "list.index.question": "Do you really want to archive the '{name}' sack?",
+  "list.index.change-type-question": "Do you really want to change the '{name}' sack to {value}?",
```
Comment (marekrozmus) [inline client/src/locales/en.json]: I don't like those `Do you really...` questions. Maybe 'Are you sure you want to...' would sound better?

File: client/src/modules/list/components/ListHeader.jsx [hunk 6]
```diff
@@ -212,6 +215,16 @@ class ListHeader extends PureComponent {
 
   hideTip = () => this.setState({ isTipVisible: false });
 
+  showDialog = event => {
+    const { value } = event.target;
+
+    this.setState({ isDialogVisible: true, newType: value });
+  };
+
+  hideDialog = () => {
```
Comment (marekrozmus) [inline client/src/modules/list/components/ListHeader.jsx]: remove braces

File: client/src/modules/list/components/ListHeader.jsx [hunk 7]
```diff
@@ -318,12 +329,37 @@ class ListHeader extends PureComponent {
...
+    const {
+      details: { name },
+      intl: { formatMessage }
+    } = this.props;
+
+    return (
+      <Dialog
+        onCancel={this.hideDialog}
+        onConfirm={this.handleChangingType}
+        pending={pendingForType}
```
Comment (adamklepacz) [inline client/src/modules/list/components/ListHeader.jsx]: pendingForTypeUpdate

**Pull request merged**